src/handle-checkout-mutation.js

```diff
@@ -1,5 +1,9 @@
 export default function handleCheckoutMutation(mutationRootKey, client) {
   return function({data, errors, model}) {
+    if (!data && errors && errors.length) {
```

Comment:
Have you thought about scenarios where we return both `data` and `errors`?

Reply:
Hi @osamie! Thanks for your review. I have not. I'm not familiar with this code and so was relying on tests to ensure existing cases were covered. I am simply a user the library and saw the TypeError (with the real error being swallowed) and thought I'd make a PR to bring attention to the issue. I am so unfamiliar with the code, that [my first attempt](https://github.com/Shopify/js-buy-sdk/commit/c39aa7fb08037f294663ba6e8b5650bbff6ab5d2) involved the assumption that if there were errors there would be no data, which a test demonstrates is not true.

#711 looks like a reasonable way to solve this (I hadn't seen that PR when I made this one.. bad job out of me!). I would be a bit hesitant about the approach to allow the data path to unfold by defaulting to an empty object. It seems to me that the ideal scenario would be to prioritize catching the invalid case and returning, rather than providing a default just to allow it to journey the data path without erring before getting to the error handling for the invalid case.

The only thing that would still be valuable from this PR is the [change I made to a test](https://github.com/Shopify/js-buy-sdk/pull/722/commits/7ceca71a18419709901d7a2cd9553f6e00269a07#diff-9400fc19b3e72a86fc71e9b744438ee8L521) since this is the test that was should have been failing given the current code, but wasn't. The fix in #711 adds its own test as well, but I think it would make sense to update the existing test so it is accurate to the scenario it is describing ("without data field").